### PR TITLE
Use AST for Prebid.js adaptor

### PR DESCRIPTION
**Pull request by hjeong12**

### CODE REVIEW

File: src/adapters/appnexus.js [hunk 1]
```diff
@@ -11,6 +11,9 @@ var bidfactory = require('../bidfactory.js');
 
 var AppNexusAdapter = function AppNexusAdapter() {
 	var isCalled = false;
+	// var AST_URL = '//acdn.adnxs.com/ast/ast.js';
+	var AST_URL ='../../../../resources_apn-seller-tag/dist/ast.js';
```
Comment (hjeong12) [inline src/adapters/appnexus.js]: I also updated ast api , so the test URL is left.

File: src/adapters/appnexus.js [hunk 2]
```diff
@@ -46,187 +49,145 @@ var AppNexusAdapter = function AppNexusAdapter() {
...
-		var sizeQueryString = utils.parseSizesInput(bid.sizes);
-		if (sizeQueryString) {
-			jptCall += sizeQueryString + '&';
-		}
-		//console.log(jptCall);
+    			apntag.anq.push(function(){
 
-		var targetingParams = utils.parseQueryStringParameters(query);
+    		        var requestTag = apntag.defineTag(astTag);
+    		        requestTag.on('adAvailable',function(){
```
Comment (hjeong12) [inline src/adapters/appnexus.js]: Using event emit for response callback.

File: src/adapters/appnexus.js [hunk 3]
```diff
@@ -46,187 +49,145 @@ var AppNexusAdapter = function AppNexusAdapter() {
...
 	}
 
-	//expose the callback to the global object:
-	pbjs.handleCB = function(jptResponseObj) {
+	pbjs.handleCB = function(targetId){
+		var placementCode = '';
 
-		if (jptResponseObj && jptResponseObj.callback_uid) {
+		//get response data
+		var data = apntag.getAdData(targetId);
```
Comment (hjeong12) [inline src/adapters/appnexus.js]: get response data using apntag.getAdData()

File: src/adapters/appnexus.js [hunk 4]
```diff
@@ -46,187 +49,145 @@ var AppNexusAdapter = function AppNexusAdapter() {
...
-
-			} else {
-				//no response data
-				// @if NODE_ENV='debug'
+
+				bid.creative_id = data.ad.creative_id;
+				bid.cpm = data.ad.cpm;
+				bid.ad = data.ad.banner.content;
+				bid.width = data.ad.banner.width;
+				bid.height = data.ad.banner.height;
```
Comment (hjeong12) [inline src/adapters/appnexus.js]: I'm not sure that Which data should be in bid response object.

File: src/adapters/appnexus.js [hunk 5]
```diff
@@ -46,187 +49,145 @@ var AppNexusAdapter = function AppNexusAdapter() {
...
+		if(!tag.sizes)
+			tag.sizes = bid.sizes;
+		tag.targetId = uuid;
 
-	};
+		return tag;
+	}
+
+	//given a starttime and an end time, hit the correct impression tracker
+	function processAndTrackLatency(startTime, endTime, placementCode) {
```
Comment (hjeong12) [inline src/adapters/appnexus.js]: should call this function?

**Comment by hjeong12** [comment]
@mkendall07  Please give me your opinion for those comments on above.

File: src/adapters/appnexus.js (hunk 2 above)
Comment (mkendall07) [inline src/adapters/appnexus.js]: we should pass an adData object back inside the handler for AST. `requestTag.on('adAvailable',function(adData){` we need to define the object first.

File: src/adapters/appnexus.js (hunk 3 above)
Comment (mkendall07) [inline src/adapters/appnexus.js]: This is ok for now until the handler passes the data

File: src/adapters/appnexus.js (hunk 4 above)
Comment (mkendall07) [inline src/adapters/appnexus.js]: This looks good to me.

File: src/adapters/appnexus.js (hunk 5 above)
Comment (mkendall07) [inline src/adapters/appnexus.js]: Yes still want to track latency

**Pull request closed without merging**